[Bug]: IsEmpty matcher doesn't work with C-style wide strings

### Describe the issue

Compiling a test with the following excerpt fails.
`ASSERT_THAT(L"NotEmpty", Not(IsEmpty()));`

```
In file included from /opt/compiler-explorer/libs/googletest/trunk/googlemock/include/gmock/gmock.h:61,
                 from <source>:1:
/opt/compiler-explorer/libs/googletest/trunk/googlemock/include/gmock/gmock-more-matchers.h: In instantiation of 'bool testing::internal::IsEmptyMatcher::MatchAndExplain(const MatcheeContainerType&, testing::MatchResultListener*) const [with MatcheeContainerType = wchar_t [9]]':
/opt/compiler-explorer/libs/googletest/trunk/googletest/include/gtest/gtest-matchers.h:647:35:   required from 'bool testing::PolymorphicMatcher<Impl>::MonomorphicImpl<T>::MatchAndExplain(T, testing::MatchResultListener*) const [with T = const wchar_t (&)[9]; Impl = testing::internal::IsEmptyMatcher]'
/opt/compiler-explorer/libs/googletest/trunk/googletest/include/gtest/gtest-matchers.h:646:10:   required from here
/opt/compiler-explorer/libs/googletest/trunk/googlemock/include/gmock/gmock-more-matchers.h:70:11: error: request for member 'empty' in 'c', which is of non-class type 'const wchar_t [9]'
   70 |     if (c.empty()) {
      |         ~~^~~~~
/opt/compiler-explorer/libs/googletest/trunk/googlemock/include/gmock/gmock-more-matchers.h:73:40: error: request for member 'size' in 'c', which is of non-class type 'const wchar_t [9]'
   73 |     *listener << "whose size is " << c.size();
      |                                      ~~^~~~
Compiler returned: 1
```

The overload of `IsEmpty` for C-style wide strings seems to be missing. The [documentation](https://github.com/google/googletest/blob/main/docs/reference/matchers.md?plain=1#L91) mentions that there should be one.

The [testcase](https://github.com/google/googletest/blob/v1.17.x/googlemock/test/gmock-matchers-misc_test.cc#L147) does not check for C-style wide string compatibility.



### Steps to reproduce the problem

See a minimal example [here](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:2,endLineNumber:11,positionColumn:2,positionLineNumber:11,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:'%23include+%3Cgmock/gmock.h%3E%0Ausing+namespace+::testing%3B%0A%0ATEST(foo,+bar)+%7B%0A++++ASSERT_THAT(L%22NotEmpty%22,+Not(IsEmpty()))%3B%0A%7D%0A%0Aint+main(int+argc,+char+**argv)+%7B%0A++::testing::InitGoogleMock(%26argc,+argv)%3B%0A++return+RUN_ALL_TESTS()%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:48.92501310959623,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:output,i:(compilerName:'x86-64+clang+14.0.0',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+ARM+GCC+11.4.0+(Compiler+%231)',t:'0')),header:(),l:'4',m:39.624724061810156,n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:armg1140,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'0',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!((name:googletest,ver:trunk)),options:'',overrides:!((name:stdver,value:c%2B%2B17)),selection:(endColumn:45,endLineNumber:3,positionColumn:45,positionLineNumber:3,selectionStartColumn:45,selectionStartLineNumber:3,startColumn:45,startLineNumber:3),source:1),l:'5',n:'0',o:'+ARM+GCC+11.4.0+(Editor+%231)',t:'0')),k:53.37353609508828,l:'4',m:60.375275938189844,n:'0',o:'',s:0,t:'0')),k:51.07498689040377,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4).


### What version of GoogleTest are you using?

Trunk was used in godbolt, so it should be a quite recent version.
I also use v1.17.x on my computer.

### What operating system and version are you using?

Windows 11 24H2

### What compiler and version are you using?

Reproducible on both gcc-11.4.0 and clang-14.

### What build system are you using?

cmake version 4.0.3

### Additional context

_No response_

--- a/googlemock/include/gmock/gmock-more-matchers.h
+++ b/googlemock/include/gmock/gmock-more-matchers.h
@@ -79,6 +79,12 @@ class [[nodiscard]] IsEmptyMatcher {
     return MatchAndExplain(std::string(s), listener);
   }
 
+  // Matches C-style wide strings.
+  bool MatchAndExplain(const wchar_t* s,
+                       MatchResultListener* listener) const {
+    return MatchAndExplain(std::wstring(s), listener);
+  }
+
   // Describes what this matcher matches.
   void DescribeTo(std::ostream* os) const { *os << "is empty"; }
 

